rfcs/0009-code-review.md

```diff
@@ -0,0 +1,182 @@
+     put into the code.
+   - The reviewee may not understand the reviewers past experience with
+     the code.
+ - [Be nice](https://qz.com/work/625870/after-years-of-intensive-analysis-google-discovers-the-key-to-good-teamwork-is-being-nice/)
+   - Read the CDS [Code of Conduct](https://github.com/cds-snc/.github/blob/master/CODE_OF_CONDUCT.md)
+ - Escalate if the review is blocked.
+   - The goal of the review is that the code looks good to the reviewer.
+     It doesn't have to be perfect, it doesn't have to be how the
+     reviewer would write it. Just, good.
+ - In general authors should merge their own PRs.
+   - This allows them to make any final touch ups or edit the commit
+     message.
+ - Cleanup branches after merge. If possible, enable the auto-delete
+   after merge option on repositories.
+ - Prefer to `Squash and merge` commits.
+   - The development history is in the PR if desired, but typically
+     doesn't add much to the main repository.
+ - Force pushing to a repo should be a last resort.
+
+### Principles for reviewees
+ - The title and description should provide all the needed context.
+   - The PR will serve as a record of the change into the future.
+   - Anything important for the reviewer to know (other approaches tried,
+     testing you’ve done, etc) is worth including.
+   - Link to any associated issues or Trello cards.
+   - If the PR is broken apart to make review easier, explain how it was
+     broken apart and include the next steps in the PR description.
+   - List any steps, caveats or gotchas that the reviewer will need to
+     know.
+ - Do your best to address all comments (even just to say "Done" or put
+   in a thumbs up).
+   - Marking the comments as `Resolved` makes it easier to understand
+     what still needs to be done.
+ - Assume the best intention from review comments
+ - Wait for the tests to pass before you merge.
+ - Tend towards smaller changes. They are easier to understand and
+   faster to review.
+
+
+### Principles for reviewers
+ - Try to comment on the PR in as short an interval is possible (<24
+   business hours preferably).
+   - The comment doesn't have to be the review, but can just be a
+     comment letting the reviewee know when you'll be able to review.
+     This gives them the opportunity to find another reviewer if needed.
+   - If you’re not sure how someone wants their code reviewed or how they
+     prefer to receive feedback, talk to them beforehand.
+ - Point out when something is particularly good or clever.
+   - Reviews often focus on the negative so highlight the good as well
+     when you see it.
+ - Be explicit.
+   - Include screenshots or gifs if they help make your point.
+ - Programming is subjective.
+   - Different approaches are valid.
+   - Ask for clarifications if you need any.
+ - Communicate which ideas you feel strongly about and those you don't.
+
+
+### Technical Considerations
+
+#### Logic
+ - Does the code work?
```

Comment:
- Are there tests that were written to cover the new logic?
- Does this introduce a certain level of tech debt?
- If there is an automated spun up instance for the PR has the functionality introduced in the PR been tested through the UI.

Reply:
Added the automated instance point to the tests section below, I think the are there tests is covered there?

Added the tech debt part here, that's a good addition.